modules/tinymce/src/themes/silver/main/ts/ui/context/ContextToolbar.ts

```diff
@@ -257,6 +262,7 @@ const register = (editor: Editor, registryContextToolbars: Record<string, Contex
       const scopes = getScopes();
       // TODO: Have this stored in a better structure
       Obj.get(scopes.lookupTable, e.toolbarKey).each((ctx) => {
+        lastToolbarKey.set(e.toolbarKey);
```

Comment:
This does look like the wrong approach, but I haven't looked into it. It does seem to be consistent with an approach we are using throughout this, however, because there are *lots* of clear statements in the above code. Sometimes, when that is the case, it can be better to group together some of them into one object ...but that all depends on whether they are set / cleared at the same time.

So, I'll have a look and see if I can find a cleaner way of achieving what you want.